### ISSUE TITLE
original_path extraction error regarding LTCurve

### ISSUE THREAD

**Issue by KaboChow** [jsvine/pdfplumber#1057]
During the process of extracting shape data from a PDF, I converted the created text letter 'o' into a shape object.

![image](https://github.com/pdfminer/pdfminer.six/assets/46550365/76ce7a9f-3c14-42eb-ada5-8eb086d78380)
Here is the curve data I obtained.

![image](https://github.com/pdfminer/pdfminer.six/assets/46550365/f9e4fe52-3f14-4ee8-aa33-6f2da9947478)
Normally, there should only be one set of curve data. 
However, it seems that there are two in this case. Here is the graphic created on the canvas using the obtained data:

![image](https://github.com/pdfminer/pdfminer.six/assets/46550365/90ce9794-8a39-4183-9c41-ea6434fe620e)
The filling color obtained for the second set of curve data is incorrect. 

This is the PDF I conducted the test on:
[LTCurve.pdf](https://github.com/pdfminer/pdfminer.six/files/13611413/LTCurve.pdf)

Is there any way to resolve this? 
Thank you very much.

**Comment by jsvine** [comment]
Hi @KaboChow, and thanks for providing this interesting example. It appears to relate to `pdfplumber`'s main dependency, [`pdfminer.six`](https://github.com/pdfminer/pdfminer.six).

It seems that there's some discussion of this general issue here: https://github.com/pdfminer/pdfminer.six/issues/861#issuecomment-1493442408

As it happens, however, [the piece of `pdfminer.six` code it likely relates to](https://github.com/pdfminer/pdfminer.six/blob/997424deaadfa23e57f15561b492c194075f950f/pdfminer/converter.py#L122-L126) is code I've contributed. Just brainstorming here, I think the issue is that folks *generally* want to decompose paths with multiple subpaths, for the purpose of rectangle detection. (See [this test for an example](https://github.com/pdfminer/pdfminer.six/blob/997424deaadfa23e57f15561b492c194075f950f/tests/test_converter.py#L107-L120).) As the issue comments above correctly point out, this makes it difficult/impossible to correctly handle more complex paths, such as shapes with holes in them. 

One solution would be to propose reverting the behavior so that it does *not* decompose complex paths, with the downside being that some clearly rectangle-like things do not get recognized as such.

Another would be to tweak the behavior so that it mostly does not decompose complex paths *except* in the case of those composed entirely of rectangles. The downside would be that this may be a confusing rule, and also that some all-rectangle complex paths are *still* intending to be understood as shapes with holes in them.

Thanks again. Will keep thinking on this, and welcome suggestions from others, too.

**Comment by KaboChow** [comment]
@jsvine Thank you for your answer.
Regarding the solution to this problem, I have done some processing on the obtained data, when the 'evenodd' value of two objects is false, to determine whether the boundaries of the two objects coincide, if they do, then the smaller side is the subpath, this method works for me, I hope it will be helpful for people who have the same confusion

**Comment by KaboChow** [comment]
Hello @jsvine, I found a problem regarding the 'evenodd' value of the object。
![image](https://github.com/jsvine/pdfplumber/assets/46550365/7eebbdda-211b-4b69-b70e-5ee8bc6ca9b2)
When I try to get the data of this porous shape, the 'evenodd' values ​​obtained are all true
![image](https://github.com/jsvine/pdfplumber/assets/46550365/afddc61a-fc7f-4261-9966-c1d07cc11c77)
This is the PDF I used for testing：
[Spin-City-Letters-6fae9bb1b9a6b3dd0f5811b066e9ed8e.pdf](https://github.com/user-attachments/files/16154938/Spin-City-Letters-6fae9bb1b9a6b3dd0f5811b066e9ed8e.pdf)

**Comment by KaboChow** [comment]
When I use letters or numbers to convert shapes, the data recognized is correct, and the value of 'evenodd' is false.
But when using a custom shape, the recognized values ​​of 'evenodd' are all true.

![image](https://github.com/jsvine/pdfplumber/assets/46550365/41934bff-cbc6-4db9-9735-51245c57cb1e)

**Comment by jsvine** [comment]
Thank you for these additional examples, @KaboChow. I'm still unsure of the best solution, given the tradeoffs described above and that any changes would have to be made to `pdfminer.six`, but these examples are still helpful.

**Comment by KaboChow** [comment]
Thank you @jsvine. The incorrect value of 'evenodd' has a great impact on my project. I have been looking for ways to solve it recently. If there is a new solution, please be sure to notify me

**Comment by dhdaines** [comment]
See discussion at pdfminer above.  The issue is that pdfminer doesn't apply any fill rules in layout analysis.  Ideally, you should be looking at the `fill` attribute, not the `evenodd` one, but it isn't getting set usefully, because pdfminer isn't applying fill rules.  As a workaround obviously you could apply the fill rule yourself ;-)

**Comment by KaboChow** [comment]
You're right @dhdaines, the "fill" property is a good way to determine whether it is a hole, but the "fill" property value of the LTCurve child object that is currently split out is inherited from the parent object, obviously the "fill" property value of the LTCurve child object is incorrect, and my ability is limited and I can't solve this problem, so I finally chose to clear the rule of splitting the LTCurve shape
